src/test/VertexUtilityTest.cpp

```diff
@@ -0,0 +1,146 @@
+ * a valid position.
+ */
+BOOST_FIXTURE_TEST_CASE(GetPosition_IndexOutOfBounds, VertexUtility)
+{
+    Vec2 v = VertexUtility::GetPosition(10, 2, 2);
+        
+    BOOST_REQUIRE_EQUAL(v.x, 0);
+    BOOST_REQUIRE_EQUAL(v.y, 5);
+}
+
+/**
+ * Tests the VertexUtility::GetPosition method for an index which is expected to be
+ * the bottom, right corner position according to the width and height.
+ */
+BOOST_FIXTURE_TEST_CASE(GetPosition_CornerIndex, VertexUtility)
+{
+    Vec2 v = VertexUtility::GetPosition(8, 3, 3);
+        
+    BOOST_REQUIRE_EQUAL(v.x, 2);
+    BOOST_REQUIRE_EQUAL(v.y, 2);
+}
+    
+/**
+ * Tests the VertexUtility::GetPosition method for a zero-index. Then the method should
+ * always return 0 independ of width and height.
+ */
+BOOST_FIXTURE_TEST_CASE(GetPosition_ZeroIndex, VertexUtility)
+{
+    for (int width = 1; width < 100; width++)
+    {
+        for (int height = 1; height < 100; height++)
+        {
+            Vec2 v = VertexUtility::GetPosition(0, width, height);
+            BOOST_REQUIRE_EQUAL(v.x, 0);
+            BOOST_REQUIRE_EQUAL(v.y, 0);
+        }
+    }
+}
+
+/**
+ * Tests the VertexUtility::GetIndexOf method for a negative x-coordinate for the position.
+ * The resulting index should be pointing to a valid entry inside of the bounds given by width
+ * and height.
+ */
+BOOST_FIXTURE_TEST_CASE(GetIndexOf_NegativePositionX, VertexUtility)
+{
+    const int width = 4;
+    const int height = 5;
+    const int index = VertexUtility::GetIndexOf(-2 * width, 0, width, height);
+    
+    BOOST_REQUIRE(index >= 0 && index < width * height);
+}
+
+/**
+ * Tests the VertexUtility::GetIndexOf method for a negative y-coordinate for the position.
+ * The resulting index should be pointing to a valid entry inside of the bounds given by width
+ * and height.
+ */
+BOOST_FIXTURE_TEST_CASE(GetIndexOf_NegativePositionY, VertexUtility)
+{
+    const int width = 4;
+    const int height = 5;
+    const int index = VertexUtility::GetIndexOf(0, -2 * height, width, height);
+    
+    BOOST_REQUIRE(index >= 0 && index < width * height);
+}
+
+/**
+ * Tests the VertexUtility::GetIndexOf method for a position outside of the bounds given by 
+ * width and height.
+ */
+BOOST_FIXTURE_TEST_CASE(GetIndexOf_OutsideOfBounds, VertexUtility)
+{
+    const int width = 4;
+    const int height = 5;
+    const int index = VertexUtility::GetIndexOf(2 * width, 2 * height, width, height);
+    
+    BOOST_REQUIRE(index >= 0 && index < width * height);
+}
+
+/**
+ * Tests the VertexUtility::GetIndexOf method for a position inside of the bounds given by
+ * width and height. The resulting index must match the scheme: index = y * width + x.
+ */
+BOOST_FIXTURE_TEST_CASE(GetIndexOf_InsideOfBounds, VertexUtility)
+{
+    const int x = 4;
+    const int y = 2;
+    const int width = x * x;
+    const int height = y * y;
+    const int index = VertexUtility::GetIndexOf(x, y, width, height);
+    
+    BOOST_REQUIRE_EQUAL(index, y * width + x);
```

Comment:
Check at least 2 indices. You can combine the index calculation and IndexOf into the check if you like. Would be more explicit